applications/GeoMechanicsApplication/custom_strategies/schemes/geomechanics_time_integration_scheme.hpp

```diff
@@ -0,0 +1,360 @@
+                                  LocalSystemVectorType& RHS_Contribution,
+                                  Element::EquationIdVectorType& EquationId,
+                                  const ProcessInfo& CurrentProcessInfo) override
+    {
+        KRATOS_TRY
+
+        rCurrentCondition.CalculateRightHandSide(RHS_Contribution, CurrentProcessInfo);
+
+        rCurrentCondition.EquationIdVector(EquationId, CurrentProcessInfo);
+
+        KRATOS_CATCH("")
+    }
+
+    void CalculateLHSContribution(Element& rCurrentElement,
+                                  LocalSystemMatrixType& LHS_Contribution,
+                                  Element::EquationIdVectorType& EquationId,
+                                  const ProcessInfo& CurrentProcessInfo) override
+    {
+        KRATOS_TRY
+
+        rCurrentElement.CalculateLeftHandSide(LHS_Contribution, CurrentProcessInfo);
+
+        rCurrentElement.EquationIdVector(EquationId, CurrentProcessInfo);
+
+        KRATOS_CATCH("")
+    }
+
+    void CalculateLHSContribution(Condition& rCurrentCondition,
+                                  LocalSystemMatrixType& LHS_Contribution,
+                                  Element::EquationIdVectorType& EquationId,
+                                  const ProcessInfo& CurrentProcessInfo) override
+    {
+        KRATOS_TRY
+
+        rCurrentCondition.CalculateLeftHandSide(LHS_Contribution, CurrentProcessInfo);
+
+        rCurrentCondition.EquationIdVector(EquationId, CurrentProcessInfo);
+
+        KRATOS_CATCH("")
+    }
+
+    void Update(ModelPart& rModelPart,
+                DofsArrayType& rDofSet,
+                TSystemMatrixType& A,
+                TSystemVectorType& Dx,
+                TSystemVectorType& b) override
+    {
+        KRATOS_TRY
+
+        int num_threads = ParallelUtilities::GetNumThreads();
+        OpenMPUtils::PartitionVector dof_set_partition;
+        OpenMPUtils::DivideInPartitions(static_cast<int>(rDofSet.size()),
+                                        num_threads, dof_set_partition);
+
+#pragma omp parallel
+        {
+            int k = OpenMPUtils::ThisThread();
+
+            typename DofsArrayType::iterator dofs_begin =
+                rDofSet.begin() + dof_set_partition[k];
+            typename DofsArrayType::iterator dofs_end =
+                rDofSet.begin() + dof_set_partition[k + 1];
+
+            // Update Displacement and Pressure (DOFs)
+            for (typename DofsArrayType::iterator itDof = dofs_begin;
+                 itDof != dofs_end; ++itDof) {
+                if (itDof->IsFree())
+                    itDof->GetSolutionStepValue() +=
+                        TSparseSpace::GetValue(Dx, itDof->EquationId());
+            }
+        }
+
+        this->UpdateVariablesDerivatives(rModelPart);
+
+        KRATOS_CATCH("")
+    }
+
+protected:
+    void CheckBufferSize(const ModelPart& rModelPart) const
+    {
+        constexpr int minimum_buffer_size = 2;
```

Comment:
In the check below, we compare an `int` with a `ModelPart::IndexType`. Perhaps it's better to use the same type:
```suggestion
        constexpr auto minimum_buffer_size = ModelPart::IndexType{2};
```

Reply:
Good point, changed